hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/payload/ExpressionPayload.scala

```diff
@@ -318,5 +305,26 @@ object ExpressionPayload {
         }
       })
   }
+
+  private val mergedSchemaCache = CacheBuilder.newBuilder().build[TupleSchema, Schema]()
+
+  def getMergedSchema(source: Schema, target: Schema): Schema = {
+
+    mergedSchemaCache.get(TupleSchema(source, target), new Callable[Schema] {
+      override def call(): Schema = {
+        val noMetaRight = HoodieAvroUtils.removeMetadataFields(target)
+        val mergedFields =
```

Comment:
`noMetaRight` -> `rightSchema`

I would suggest you keep the `mergeSchema` method, and `getMergedSchema` can call this method to make the logic more clear.

Reply:
got it